File: SMM.IServices/Models/Post/PostModel.cs
using SMM.IServices.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Web;

namespace SMM.IServices.Models.Post
{
    public class PostModel
    {
        /// <summary>
        /// Ид
        /// </summary>
        public int Id { get; set; }
        /// <summary>
        /// Название поста
        /// </summary>
        public string Name { get; set; }
        /// <summary>
        /// Содержание поста
        /// </summary>
        public string Content { get; set; }
        /// <summary>
        /// Дата создание
        /// </summary>
        public DateTime DateCreate { get; set; }
        /// <summary>
        /// Дата публикации в соц сетях
        /// </summary>
        public DateTime DatePublic { get; set; }
        /// <summary>
        /// Ид проекта
        /// </summary>
        public int ProjectId { get; set; }
        /// <summary>
        /// Ид пользователя создавшего пост
        /// </summary>
        public int UserId { get; set; }
        /// <summary>
        /// Статус пользователя
        /// </summary>
        public EnumStatusPost Status { get; set; }
        #region Файлы 
        /// <summary>
        /// Файл изображения (для сохранения)
        /// </summary>
        public HttpPostedFileBase ImageFile { get; set; }
        /// <summary>
        /// Файл с видео
        /// </summary>
        public HttpPostedFileBase VideoFile { get; set; }
        #endregion
        /// <summary>
        /// Ссылки на фотографию
        /// </summary>
        public List<string> ImagesUrl { get; set; }
       
        /// <summary>
        /// Комментарии 
        /// </summary>
        public List<CommentModel> Comments { get; set; }
        /// <summary>
        /// Ид поста в одноклассниках
        /// </summary>
        public string PostIdOK { get; set; }
    }
}

File: SMMService/asserts/js/Post/PostModel.js
var Post = Post || {};

(function () {

    Post.PostModel = function (theParams) {
        theParams = theParams || {};
        this.Id = ko.observable(theParams.Id || 0);
        this.Name = ko.observable(theParams.Name || "");
        this.Content = ko.observable(theParams.Content || "");
        this.DatePublic = ko.observable(theParams.DatePublic || "");
        this.DatePublicTime = ko.observable(theParams.DatePublicTime || "");
        this.Status = ko.observable(theParams.Status || "");
        this.ImageFile = ko.observable(theParams.ImageFile || "");
        this.VideoFile = ko.observable(theParams.VideoFile || "");
        this.ImagesUrl = ko.observableArray(theParams.ImagesUrl || "");
        this.Comments = ko.observableArray(theParams.Comments ? theParams.Comments.map(function (item) { return new Post.CommentModel(item) }) : []);

        this.PostIdOK = theParams.PostIdOK || "";
        this.PostIdVK = theParams.PostIdVK || "";

        return this;
    };

    /**
    * Определяем конструктор
    */
    Post.PostModel.prototype.constructor = Post.PostModel;

    Post.PostModel.prototype.GetData = function () {
        return {
            Id: this.Id(),
            Name: this.Name(),
            Content: this.Content(),
            DatePublic: new Date(this.DatePublic() + " " + this.DatePublicTime()).toISOString(),
            ImageFile: this.ImageFile(),
            VideoFile: this.VideoFile(),
        };
    }

    Post.PostModel.prototype.GetFormData = function () {
        var formData = new FormData();
        formData.append("Id", this.Id());
        formData.append("Name", this.Name());
        formData.append("Content", this.Content());
        formData.append("DatePublic", new Date(this.DatePublic() + " " + this.DatePublicTime()).toISOString());
        formData.append("ImageFile", this.ImageFile());
        formData.append("VideoFile", this.VideoFile());
        return formData;
    }

    
})();